### ISSUE TITLE
List of packages failing World build on RV32

### ISSUE THREAD

**Issue by kraj** [riscv/meta-riscv#202]
This is a placeholder for tracking world build failures on rv32, the current list and respective error logs are here [1]

[1] http://errors.yoctoproject.org/Errors/Build/97499/

**Comment by pino-kim** [comment]
@kraj ,
hmm.....
I want to get samething in my bild system.
just build below command?
                 MACHINE=qemuriscv32 bitbkae core-image-full-cmdline

**Comment by kraj** [comment]
@pino-kim you will run `bitbake -k world`

**Comment by alistair23** [comment]
That's a great list!

I had a quick look and it looks like a lot of these just don't have RV32 support yet, so this will be a fair amount of work.

**Comment by kraj** [comment]
@alistair23  I think some of the packages fail due to syscalls e.g.
```
error: 'SYS_mmap' was not declared in this scope
error: 'SYS_futex' was not declared in this scope
```
These perhaps are fixable easily

some are atomics related can be fixed by adding -pthread to linker cmdline

a big category is then of the not ported ones.

**Comment by alistair23** [comment]
Cool! Thanks for looking into this.

**Comment by ajf58** [comment]
This has a label of "help wanted", is that still the case? If so, what additional help is required?

**Comment by kraj** [comment]
@ajf58 yes perhaps fixing the 64bit time_t issues for rv32 is a  big one and also some key packages like
libseccomp, libunwind are key elements needed to enable various yocto features.

**Comment by sanjay-embedded** [comment]
@kraj ,

Do we still have issue with world build ?
Or it's quite stable.

Any specific points where we can contribute for risc-v ?
